src/scanner/nftOwnersAtBlock.js

```diff
@@ -0,0 +1,112 @@
+const { Api } = require('@cennznet/api');
+const { saveListingData, saveWalletData, updateProcessedBlockInDB, updateFinalizedBlock } = require('./utils');
+const mongoose = require('mongoose');
+const { NftWallet, NftListing  } = require('../mongo/models');
+require("dotenv").config();
+const logger = require('../logger');
+
+async function main () {
+    // Create the API and wait until ready
+    const api = await Api.create({network: 'nikau',});
+
+    const connectionStr = `mongodb://${process.env.MONGO_INITDB_ROOT_USERNAME}:${process.env.MONGO_INITDB_ROOT_PASSWORD}@${process.env.MONGO_HOST}:${process.env.MONGO_PORT}/admin`;
+    await mongoose.connect(connectionStr);
+    // get current block
+    let signedBlock = await api.rpc.chain.getBlock();
+
+    // get current block height and hash
+    let currentHeight = signedBlock.block.header.number;
+    const blockHash = signedBlock.block.header.hash;
+    await updateProcessedBlockInDB(currentHeight, currentHeight);
+
+    const entries = await api.query.nft.tokenOwner.entries();
+    logger.info(`Got all token entries`);
+    const users = [];
+    await Promise.all(
+        entries.map(async (detail) => {
+            const owner = detail[1].toString();
+            users.push(owner);
+            const [collectionId, seriesId] = detail[0].toHuman()[0];
+
+            const serialNumber = (detail[0].toHuman()[1]).replace(/,/g, '');
+            await saveWalletData(owner, [], [[collectionId, seriesId, serialNumber]]);
+        })
+    );
+
+    const userUniqueList = new Set(users);
+    const userList = Array.from(userUniqueList);
+    logger.info('NFT WALLET VERIFICATION...');
```

Comment:
why do we need the verification steps? how can it fail? should we continue if it does?

Reply:
no.. this never fails.. had added for my satisfaction to check all the records gets in db... I can remove/comment it